### PR TITLE
Allow query for all openstack info with spec in elastic

### PR DESCRIPTION
**Pull request by adrianfusco**
Ask openstack information for a single job using the elasticsearch source. Example of usage: OSPCRE-356

### CODE REVIEW

File: tests/unit/sources/elasticsearch/test_api.py [hunk 1]
```diff
@@ -421,6 +423,11 @@ def test_get_deployment(self: object, mock_query_hits: object) -> None:
         self.assertEqual(deployment.ip_version.value, '4')
         self.assertEqual(deployment.topology.value, '')
 
+    def test_spec_deployment(self: object):
```
Comment (jsanemet) [inline tests/unit/sources/elasticsearch/test_api.py:426]: This tests the error that happens when there are more than on job, but what about the 'for' loop that follows it. That looks more important to test.

**Review by jsanemet — APPROVED** [review]
Just left a small comment. Not enough to require changes, but take a look.

File: cibyl/plugins/openstack/sources/elasticsearch.py [hunk 2]
```diff
@@ -127,6 +128,26 @@ def append_get_specific_field(field: str):
...
+            'ip_version',
+            'dvr',
+            'network_backend',
+            'storage_backend',
+            'osp_release'
+        ]
+
+        if 'spec' in kwargs:
+            if len(jobs_found) > 1:
+                raise ElasticsearchException(
```
Comment (cescgina) [inline cibyl/plugins/openstack/sources/elasticsearch.py:142]: This exception should inherit from SourceException

**Review by cescgina — CHANGES_REQUESTED** [review]
Minor issue but the ElasticSearchException does not inherit from SourceException, which causes it to not follow the exception format

**Review by cescgina — APPROVED** [review]
LGTM, let's unify the spec erros of elasticsearch and Jenkins in another PR, good job!

**Pull request merged**